src/server.rs

```diff
@@ -740,7 +746,11 @@ impl LanguageServer for LspServer {
             })
         }
 
-        Ok(Some(results))
+        Ok(if results.is_empty() {
+            None
+        } else {
+            Some(results)
+        })
```

Comment:
Is this just a (nice but) unrelated set of changes?

Reply:
It ensured that the tests still returned `None` instead of an empty `Vec` (or other empty type).